modin/logging/logger_function.py

```diff
@@ -62,19 +119,20 @@ def run_and_log(*args, **kwargs):
             Any
             """
             if LogMode.get() == "disable":
-                return f(*args, **kwargs)
+                return obj(*args, **kwargs)
 
             logger = get_logger()
+            logger_level = getattr(logger, log_level)
+            logger_level(start_line)
             try:
-                logger_level = getattr(logger, log_level.lower())
-            except AttributeError:
-                raise AttributeError(f"Invalid log_level: {log_level}")
-
-            logger_level(f"START::{modin_layer.upper()}::{function_name}")
-            result = f(*args, **kwargs)
-            logger_level(f"STOP::{modin_layer.upper()}::{function_name}")
+                result = obj(*args, **kwargs)
+            except BaseException:
+                logger.exception(stop_line)
+                raise
+            logger_level(stop_line)
             return result
 
-        return run_and_log
+        # make sure we won't decorate multiple times
+        return disable_logging(run_and_log)
```

Comment:
I don't think I fully understand why this bit of code is necessary. Could you explain a bit further? I don't think it necessarily needs more commenting, more just for my own clarity.

Reply:
it isn't exactly necessary I would say, more like a safety net (example - someone might try to decorate both a class and a method - maybe I should warn instead of ignoring double decoration?..)

Reply:
I see, so this is more of a developer check to ensure no double decoration. I think the existing approach is fine in that case then.